tss-esapi/src/structures/pcr/select.rs

```diff
@@ -27,10 +27,10 @@ pub struct PcrSelect {
 
 impl PcrSelect {
     /// Creates a new PcrSelect
-    pub fn new(size_of_select: PcrSelectSize, pcr_slots: &[PcrSlot]) -> Self {
+    pub fn new(size_of_select: PcrSelectSize, mut pcr_slots: Vec<PcrSlot>) -> Self {
         PcrSelect {
             size_of_select,
-            selected_pcrs: pcr_slots.iter().cloned().collect(),
+            selected_pcrs: pcr_slots.drain(..).collect(),
```

Comment:
I wasn't sure about this one, whether `Vec` is a fair replacemenet or whether I should use something else, maybe something generic?

Reply:
hrrmm, that is weird it is only a value.. should be no need to clone it.

Reply:
Before we used to get `&[PcrSlot]` - so we had to clone the stuff behind the reference.

Reply:
This might work:
```rust
    /// Creates a new PcrSelect
    pub fn new(size_of_select: PcrSelectSize, pcr_slots: &[PcrSlot]) -> Self {
        PcrSelect {
            size_of_select,
            selected_pcrs: pcr_slots
                .iter()
                .fold(BitFlags::<PcrSlot>::empty(), |mut acc, &v| {
                    acc.insert(v);
                    acc
                }),
        }
    }
```

Reply:
True, but that's essentially doing a (potentially more expensive) version of `cloned`, by hand

Reply:
Oh, that's literally what's happening under the hood with the `.collect()` implementations: https://docs.rs/enumflags2/0.7.3/src/enumflags2/lib.rs.html#853

Reply:
Yeah, it is a set under the hood. Would be nice to be able to it without cloning. Either using from_iterator or collect.

Reply:
```
    /// Creates a new PcrSelect
    pub fn new(size_of_select: PcrSelectSize, pcr_slots: &[PcrSlot]) -> Self {
        PcrSelect {
            size_of_select,
            selected_pcrs: BitFlags::<PcrSlot>::from_iter(pcr_slots.iter().copied()),
        }
    }
```

```
    /// Creates a new PcrSelect
    pub fn new(size_of_select: PcrSelectSize, pcr_slots: &[PcrSlot]) -> Self {
        PcrSelect {
            size_of_select,
            selected_pcrs: pcr_slots.iter().copied().collect(),
        }
    }
```

Both works and are better then the previous.